pkg/config/config.go

```diff
@@ -64,12 +78,21 @@ type Config struct {
 
 // Init takes the command line argumenst from the urfave/cli context and composes the configuration
 func NewConfig(ctx *cli.Context) (*Config, error) {
-	c := &Config{}
-	c.apply(ctx)
 
-	// TODO: work on reading config file from custom path/file (reproducibility)
-	// 		 export the current conf into a file?
-	return c, nil
+	if ctx.IsSet("default-config") {
```

Comment:
The ConfigStruct should be only modified by `cmd`. This way, you can parse the configuration struct in the same way always, and parse the `CidSurce` later with the function `find_cid_source(conf)` that you defined